hack/kind-cluster-v1.12.8.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+set -e
+
+usage() {
+    cat <<EOF
+This script use kind to create Kubernetes cluster,about kind please refer: https://kind.sigs.k8s.io/
+Before run this script,please ensure that:
+* have installed docker
+* have installed golang (version >= 1.11)
+* have install helm / kubectl
+
+Options:
+       -n,--name               Name of the Kubernete cluster,default value: kind
+       -c,--nodeNum            the count of the cluster nodes,default value: 6
+       -v,--k8sVersion        version of the Kubernetes cluster,default value: v1.12.8
+Usage:
+    $0 --name testCluster --nodeNum 4 --k8sVersion v1.12.9
+EOF
+    exit 1
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    -n|--name)
+    clusterName="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -c|--nodeNum)
+    nodeNum="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -v|--k8sVersion)
+    k8sVersion="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    *)    # unknown option
+    echo "unknown option: $key"
+    usage
+    exit 1
+    ;;
+esac
+done
+
+clusterName=${clusterName:-kind}
+nodeNum=${nodeNum:-6}
+k8sVersion=${k8sVersion:-v1.12.8}
+
+echo "clusterName: ${clusterName}"
+echo "nodeNum: ${nodeNum}"
+echo "k8sVersion: ${k8sVersion}"
+
+echo "############ install kind ##############"
+if hash kind 2>/dev/null;then
+    echo "kind have installed"
+else
+    echo "start install kind"
+    GO111MODULE="on" go get sigs.k8s.io/kind@v0.4.0
+    echo PATH=${PATH}:$(go env GOPATH)/bin >> ${HOME}/.profile && source ${HOME}/.profile
```

Comment:
It's not a good idea to touch users' profile file and source that file again. Shell environment is complex, as for `bash`, if the file `~/.bash_profile` exists, `~/.profile` will be not read. If the shell is invoked as an interactive non-login shell, the `~/.profile` will not be read too. See the [docs about bash startup files](https://www.gnu.org/software/bash/manual/html_node/Bash-Startup-Files.html).

My suggestion is to simply ask the users to install kind here.

Optional, we can check our requirements at the beginning of the script:

- kind 0.4.0+
- helm
- kubectl